src/local-history/local-history-manager.ts

```diff
@@ -229,4 +229,36 @@ export class LocalHistoryManager {
             });
         }
     }
+
+    /**
+     * Removes all revisions of the active file.
+     * @param uri: the uri of the active file.
+     */
+    public clearHistory(uri: vscode.Uri): void {
+        if (uri) {
+            const hashedEditorPath = checksum(uri.fsPath);
+            const hashedFolderPath = path.join(os.homedir(), '.local-history', hashedEditorPath);
+
+            // Check if local history exists for the active file
+            if (!fs.existsSync(hashedFolderPath)) {
+                vscode.window.showInformationMessage(`No local history file found for '${path.basename(uri.fsPath)}'`);
+                return;
+            }
+
+            try {
+                vscode.window.showWarningMessage(`Are you sure you want to delete all revisions for '${path.basename(uri.fsPath)}' permanently?`, { modal: true }, 'Delete').then((selection) => {
+                    if (selection === 'Delete') {
+                        const files = fs.readdirSync(hashedFolderPath);
+                        for (const file of files) {
+                            fs.unlinkSync(path.join(hashedFolderPath, file));
+                        }
+                        fs.rmdirSync(hashedFolderPath);
```

Comment:
[question] we can use a library to recursively remove it [rimraf](https://github.com/isaacs/rimraf) which replicates the behavior of `rm -rf` as done in UNIX. Do you think it would be neater to use that?

Reply:
@Anasshahidd21 I did actually look into `rimraf` and `fs-extra` yesterday to see if they are good alternatives for removing directories that are not empty. IMO, I think we should try to stick with nodeJS core lib when dealing with path since 3rd party package might be unmaintained in the future.

Reply:
Looking at the documentation of node js, `fs` API. It does have a `recursive` flag for `rmdir` but it is not stable yet, so sticking to core lib is better for now.